app/models/communitarian/resolution.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require_relative "../../jobs/regular/reopen_resolution"
+require_relative "../post"
+
+module Communitarian
+  class Resolution
+    using Communitarian
+
+    def self.reopened_poll_post_attributes
+      %w[user_id topic_id raw post_type last_editor_id last_version_at user_deleted public_version]
+    end
+
+    def self.reopen_weekly_resolution!(original_post)
+      original_poll = original_post.polls.first
+      return if original_post.user_deleted || !original_post.resolution?
+
+      post_attributes = original_post.attributes.slice(*self.reopened_poll_post_attributes)
+      post = Post.create!(post_attributes)
+      poll = post.reload.polls.first
+      poll.update!(close_at: self.next_close_time, is_weekly: true)
+      self.schedule_jobs(post)
+    end
+
+    def self.schedule_jobs(post)
+      ::DiscoursePoll::Poll.schedule_jobs(post)
+      return unless post.resolution?
+
+      job_args = { post_id: post.id }
+      Jobs.cancel_scheduled_job(:reopen_resolution, job_args)
+      Jobs.enqueue_at(self.next_reopen_time(post.polls.first.close_at), :reopen_resolution , job_args)
+    end
+
+    def self.next_close_time(today = self.current_time)
+      next_close_day = if (self.close_week_day - today.wday) % 7 < 2
```

Comment:
Are you sure that this method work correctly?
I think that we need to find next week day that will be saved in `self.close_week_day`
```
next_close_day = today.beginning_of_week(self.close_week_day).midnight
next_close_day += 1.week if next_close_day.past?
next_close_day.change(hour: self.close_hour)
```

Reply:
I've refactored it, now I hope you'll admit method correctness